src/main/java/uk/gov/hmcts/divorce/citizen/notification/conditionalorder/AppliedForConditionalOrderNotification.java

```diff
@@ -28,23 +49,104 @@
+                partnerTemplateVars(data, id, data.getApplicant2(), data.getApplicant1(), APPLICANT1),
+                data.getApplicant2().getLanguagePreference()
+            );
+        }
     }
 
-    private Map<String, String> templateVars(CaseData caseData, Long id, Applicant applicant, Applicant partner) {
+    private Map<String, String> templateVars(CaseData caseData, Long id, Applicant applicant, Applicant partner, String whichApplicant) {
         Map<String, String> templateVars = commonContent.mainTemplateVars(caseData, id, applicant, partner);
-        templateVars.put(CO_REVIEWED_BY_DATE,
-            caseData.getConditionalOrder()
-                .getConditionalOrderApplicant1Questions()
-                .getSubmittedDate()
-                .plusDays(21).format(DATE_TIME_FORMATTER));
+        templateVars.put(PLUS_21_DUE_DATE,
+            coQuestions(caseData, whichApplicant).getSubmittedDate().plusDays(21).format(DATE_TIME_FORMATTER));
+        if (!caseData.getApplicationType().isSole()) {
+            templateVars.putAll(jointTemplateVars(caseData, partner, whichApplicant));
+        }
         return templateVars;
     }
+
+    private Map<String, String> partnerTemplateVars(CaseData data, Long id, Applicant applicant, Applicant partner, String whichPartner) {
+        Map<String, String> templateVars = commonContent.mainTemplateVars(data, id, applicant, partner);
+        templateVars.put(PLUS_14_DUE_DATE,
+            coQuestions(data, whichPartner).getSubmittedDate().plusDays(14).format(DATE_TIME_FORMATTER));
+        return templateVars;
+    }
+
+    private Map<String, String> jointTemplateVars(CaseData data, Applicant partner, String whichApplicant) {
+        boolean partnerApplied = alreadyApplied(data, whichPartner(whichApplicant));
+        Map<String, String> templateVars = new HashMap<>();
+        templateVars.put(WIFE_APPLIED, yesNo(partnerApplied && data.isDivorce() && Gender.FEMALE.equals(partner.getGender())));
+        templateVars.put(HUSBAND_APPLIED, yesNo(partnerApplied && data.isDivorce() && Gender.MALE.equals(partner.getGender())));
+        templateVars.put(CIVIL_PARTNER_APPLIED, yesNo(partnerApplied && !data.isDivorce()));
+        templateVars.put(PARTNER_APPLIED, yesNo(partnerApplied));
+        templateVars.put(WIFE_DID_NOT_APPLY,
+            yesNo(!partnerApplied && data.isDivorce() && Gender.FEMALE.equals(partner.getGender())));
+        templateVars.put(HUSBAND_DID_NOT_APPLY,
+            yesNo(!partnerApplied && data.isDivorce() && Gender.MALE.equals(partner.getGender())));
+        templateVars.put(CIVIL_PARTNER_DID_NOT_APPLY, yesNo(!partnerApplied && !data.isDivorce()));
+        templateVars.put(PARTNER_DID_NOT_APPLY, yesNo(!partnerApplied));
+        templateVars.put(PARTNER_DID_NOT_APPLY_DUE_DATE,
+            !partnerApplied ? coQuestions(data, whichApplicant).getSubmittedDate().plusDays(14).format(DATE_TIME_FORMATTER) : "");
+        return templateVars;
```

Comment:
too many conditions and methods making it difficult to read. Would it be possible to make it simpler ?